wfl/calculators/generic.py

```diff
@@ -57,14 +57,15 @@ def run_op(atoms, calculator, properties=None, output_prefix='_auto_', verbose=F
             # don't have that as default
             at.calc.calculate(at, properties=properties, system_changes=all_changes)
             calculation_succeeded = True
+            if f'{output_prefix}calculation_failed' in at.info:
+                del at.info[f'{output_prefix}calculation_failed']
         except Exception as exc:
             warnings.warn(f'calculation failed with exception {exc}')
+            at.info[f'{output_prefix}calculation_failed'] = True
```

Comment:
I don't love this way of doing error passing. For one, the calling function doesn't necessarily know the name of the info key where the failed status will be stored, since the default `output_prefix` will end up as different things. In addition, the particular exception is only saved in stderr (via warnings), and it might be nice to have it saved.  I'm not sure what to do about the first issue - maybe there's no good alternative, and I guess we're already accepting that the results are saved in keys that aren't necessarily known by the calling routine (so maybe eliminate `_auto_`?).  What do you think about saving the exception (convert to string) is saved in the `{output_prefix}calculation_failed` key?  Also, should we have an optional argument (default False?) which can be used to instead of ignoring it, actually raise the exception?

Reply:
If there's ever need for the calling function to know `output_prefix`, it should simply be specified? I never leave it to `_auto_` and would be for making `output_prefix` mandatory to specify. 

Here I have copied what's in all the calculators (e.g. line 108 of castep.py), and I see it as more of a convenience flag rather than a way to pass errors? I don't really see a problem with the exception being written to stderr/warnings and can't really imagine needing it associated with `Atoms`. Maybe in more complicated workflows, but for me it would be enough to spot that some calculations have failed, check the reason in the output, fix them and re-submit just the configs I need. I think having the exceptions in `info` would be a bit cumbersome and more often than not will have to then be deleted. 

I thought I've seen calculator exceptions being optionally raised, but maybe I'm mistaking that with `skip_failed` of `iterable_loop`. I've just added a `raise_calc_exceptions` flag for this.

Reply:
Let's merge this now, but think in the long term about wether there's a better way to do this.